src/manage_passphrase.rs

```diff
@@ -69,6 +57,43 @@ ffi!(
     }
 );
 
+pub fn encrypt_cert(cert: Cert, passphrase: *const c_char) -> Result<Cert> {
+    let have_passphrase = unsafe { passphrase.as_ref() }.is_some();
+    if !have_passphrase {
+        // nothing to do
+        Ok(cert)
+    } else {
+        let passphrase = unsafe { check_cstr!(passphrase) }
+            .to_str()
+            .map_err(|_| illegal_value("new passphrase cannot be converted to string"))?;
+
+        let passphrase = Password::from(passphrase);
+        let encrypted_packets = encrypted_packets(&cert, &passphrase)?;
```

Comment:
```suggestion
        let encrypted_packets = encrypt_packets(&cert, &passphrase)?;
```

Same for `decrypted_packets`.

I find the naming confusing. Why is it `encrypt_cert()` but `encrypted_packets()`? Both functions do the same thing, just on different objects. So the naming should be aligned.

(I know it was like this before, but it would be an opportunity to fix it.)

Reply:
No, not on different objects, but result and behavior is different. From the naming alone, `encrypt_cert` indicates that a cert gets encrypted, and `encrypted_packets` yields the encrypted packets of something.

Reply:
But `encrypt_cert` yields the encrypted cert, so let's call it `encrypted_cert`.
And `encrypted_packets` encrypts the packets, so let's call it `encrypt_packets`.

=> I still don't see the difference. Both functions encrypt X and then they yield the encrypted X. Both functions could reasonably be called `encrypt_X` and `encrypted_X`. So avoid confusion (I'm confused) we should use the same naming scheme.

Reply:
It's natural language. `encrypt_cert` (verb, object) is a _transformation_. The original certificate may be consumed or not considered important anymore. `encrypted_packets` is a _getter_. In java I may have called it `getEncryptedPackets` and put it on a certificate. Wait, no, I don't think java allows such monkey patching. The object it operates on (certificate) is not touched by the operation. I can use `get_encrypted_packets_from_certificate` and the like if that helps you. Please also note that `encrypt_cert` is public, the other is purely internal at the moment.

Reply:
Ah, I missed that one is `&Cert` and the other one is `Cert`. Now it makes more sense.